### PR TITLE
start using zio mock

### PR DESCRIPTION
**Pull request by BalduinLandolt**
### Task Description/Number

Continuation of our mob-session on testing, where I replaces the existing hand-made mocks with ZIO Mock

### Basic Requirements

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

- [ ] Bugfix: represents bug fixes
- [ ] Refactor: represents production code refactoring
- [ ] Feature: represents a new feature
- [ ] Documentation: documentation changes (no production code change)
- [ ] Chore: maintenance tasks (no production code change)
- [ ] Style: styles updates (no production code change)
- [x] Test: all about tests: adding, refactoring tests (no production code change)
- [ ] Other... Please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### Does this PR change client-test-data?

- [ ] Yes (don't forget to update the JS-LIB team about the change)
- [x] No

### CODE REVIEW

File: webapi/src/test/scala/org/knora/webapi/responders/admin/ProjectsServiceLiveSpec.scala [hunk 1]
```diff
@@ -1,91 +1,177 @@
...
-  final case class BridgeMock() extends ActorToZioBridge {
-    var msg: ResponderRequest = null
-
-    override def askAppActor[R: Tag](message: ResponderRequest): Task[R] = {
-      msg = message
-      println("msg", msg)
-      println("type", msg.getClass())
-      ZIO.attempt(null.asInstanceOf[R])
-    }
+  val uuidGenerator = new UuidGenerator {
```
Comment (irinaschubert) [inline webapi/src/test/scala/org/knora/webapi/responders/admin/ProjectsServiceLiveSpec.scala:39]: We have a UuidGeneratorMock that you could use here. You can have a look at `UuidGeneratorMockSpec` to see how to use it.

File: webapi/src/test/scala/org/knora/webapi/responders/admin/ProjectsServiceLiveSpec.scala [hunk 2]
```diff
@@ -1,91 +1,177 @@
...
+    override def createRandomUuid: UIO[UUID] = for {
+      random <- ZIO.random
+      _      <- random.setSeed(0) // makes the random deterministic: bb20b45f-d4d9-4138-bd93-cb799b3970be
+      uuid   <- random.nextUUID
+    } yield uuid
```
Comment (seakayone) [inline webapi/src/test/scala/org/knora/webapi/responders/admin/ProjectsServiceLiveSpec.scala:44]: See my comment above as to why I think the `UuidGenerator` should be deprecated with this approach of managing randomness.

**Review by irinaschubert — APPROVED** [review]
LGTM

**Review by seakayone — APPROVED** [review]

File: webapi/src/test/scala/org/knora/webapi/responders/admin/ProjectsServiceLiveSpec.scala (hunk 1 above)
Comment (irinaschubert) [reply webapi/src/test/scala/org/knora/webapi/responders/admin/ProjectsServiceLiveSpec.scala:39]: Ah, I've just read @seakayone 's comment below. In that case we can (and should) remove our UuidGenerator.

Comment (BalduinLandolt) [reply webapi/src/test/scala/org/knora/webapi/responders/admin/ProjectsServiceLiveSpec.scala:39]: see Christian's comment below :)

**Comment by BalduinLandolt** [comment]
@subotic correct. We were doing some experiments here, and we have agreed to merge it like that (after some clean-up) because this is not productive code at the moment, so it can't hurt to have it.  
But this led to a discussion about how we want our test strategy to look exactly, which I will turn into an RFC. There we'll be happy to have your input too, if you like. And once the the consensus is explicit, I'll turn it into an ADR, so that it's clear for everyone.

File: webapi/src/test/scala/org/knora/webapi/routing/admin/ProjectsRouteZSpec.scala [hunk 3]
```diff
@@ -39,332 +41,152 @@ object ProjectRouteZSpec extends ZIOSpecDefault {
...
-  /**
-   * Common layers and expectations for ActorToZioBridge mock
-   */
-  // private val commonLayers: URLayer[ActorToZioBridge, ProjectsRouteZ] =
-  //   ZLayer.makeSome[ActorToZioBridge, ProjectsRouteZ](
-  //     AppConfig.test,
-  //     ProjectsRouteZ.layer,
-  //     RestProjectsService.layer,
-  //     authenticationMiddlewareLayer
-  //   )
```
Comment (mpro7) [inline webapi/src/test/scala/org/knora/webapi/routing/admin/ProjectsRouteZSpec.scala:81]: Do we still need all the commented out code in this file?

**Review by mpro7 — APPROVED** [review]
Just one comment regarding commented out comments, see below.

File: webapi/src/main/scala/org/knora/webapi/responders/admin/ProjectsService.scala [hunk 4]
```diff
@@ -23,7 +21,7 @@ trait ProjectsService {
   ): Task[ProjectOperationResponseADM]
 }
 
-final case class ProjectsServiceLive(bridge: ActorToZioBridge) extends ProjectsService {
+final case class ProjectsServiceLive(bridge: ActorToZioBridge, uuidGenerator: UuidGenerator) extends ProjectsService {
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply webapi/src/main/scala/org/knora/webapi/responders/admin/ProjectsService.scala:24]: you're right, I remove it here (but will leave it in in the other places it's used, as that's outside of the scope of this task)

File: webapi/src/test/scala/org/knora/webapi/routing/admin/ProjectsRouteZSpec.scala (hunk 3 above)
Comment (BalduinLandolt) [reply webapi/src/test/scala/org/knora/webapi/routing/admin/ProjectsRouteZSpec.scala:81]: not really. But when this PR is merged into the base branch (from our mob session) I'll have to do a clean-up pass anyways, so I think I'll do all the clean-up at once

**Pull request merged**